.github/workflows/lint.yaml

```diff
@@ -15,18 +15,21 @@ jobs:
       - name: Check out the repository
         uses: actions/checkout@v4
 
-      - name: Set up Python
-        uses: actions/setup-python@v5
+      - name: Install the latest version of uv and set the python version to 3.13
+        uses: astral-sh/setup-uv@v6
         with:
-          python-version: '3.11'
+          python-version: 3.13
+          activate-environment: true
 
       - name: Install pre-commit
-        run: pip install pre-commit
+        run: uv sync --extra lint
 
       - uses: actions/cache@v4
```

Comment:
Is this cache still needed? It seems that if we use UV we should follow:
https://docs.astral.sh/uv/guides/integration/github/#caching